R/pin-read-write.R

```diff
@@ -111,6 +116,17 @@ pin_write <- function(board, x,
   )
   meta$user <- metadata
 
+  if (check_hash) {
+    old_hash <- possibly_pin_meta(board, name)$pin_hash
```

Comment:
Why do we think this might fail?

Reply:
This is to let someone use `check_hash = TRUE` when they write a pin for the first time, before there is any metadata. `old_hash` returns as `NULL` here when there isn't any metadata.

Reply:
Hmmm, I think I'd prefer an explicit `pin_exists()` check? Is there some reason not to do this?